### DISCUSSION TITLE
Shared components in Structurizr for Java library

### DISCUSSION THREAD

**Discussion by prasadvp** [structurizr/java#246] [Q&A]
We have `include` tag in DSL that allows to create shared components and use it elsewhere. However using Java library, any software system can be created only with workspace which makes it impossible to create shared components. 
I am thinking of creating a wrapper that exposes function to take a software system as an input in the model class. 
Please let me know if there is an out of box option available for creating and using shared components using Java library.

**Comment by moarsoarse** [comment]
In my case I need to include shared library into multiple containers. So I just wrote a fabric method:

```
    static Container prepareMicroservice(SoftwareSystem platform, String name, String description, String technology){
        Container microservice = platform.addContainer(name,description,technology);
        //Shared library components
        return microservice;
    }
```

**Reply by prasadvp** [reply]
Thank you for responding. Yes, this works for `containers`, `components`.

**Comment by prasadvp** [comment]
I believe this may work. 
```
public class WorkspaceWrapper {
      private Workspace workspace = new Workspace("Getting Started", "This is a model of my software system.");
      public static Model model = workspace.getModel();
}
---------------------------

import WorkspaceWrapper;
public class GettingStarted {
   SoftwareSystem softwareSystem = WorkspaceWrapper.model.addSoftwareSystem("Software System", "My software system."); 
}
```
